tests: fix flaky test - tests/unit/test_workflow_files.py

Two of the unit tests are flaky:

```
FAILED tests/unit/test_workflow_files.py::test_install_workflow__max_depth[foo/foo/-False]
FAILED tests/unit/test_workflow_files.py::test_install_workflow__next_to_flow_file[None-None]
```

This is because of interaction with symlink dirs:

```
cylc.flow.exceptions.WorkflowFilesError: Symlink dir target already exists: (/var/tmp/pytest-of-osanders/pytest-3152/popen-gw0/test_install_workflow__next_to2/cylc-run/faden/run1/log ->) /data/users/osanders/cylc-run/faden/run1/log
E           Tip: in future, use 'cylc clean' instead of manually deleting workflow run dirs.
```

The unit tests should not be creating workflows in the `cylc-run` directory directly. If needed they should be installed into a test hierarchy as the integration and functional tests are.

Perhaps these tests might be more appropriate as integration tests?

**Pull requests welcome!**
<!-- This is an Open Source project - please consider contributing code yourself
(please read `CONTRIBUTING.md` before starting any work though). -->

--- a/tests/unit/conftest.py
+++ b/tests/unit/conftest.py
@@ -204,3 +204,11 @@ def xtrigger_mgr() -> XtriggerManager:
             create_autospec(Scheduler, workflow=workflow_name, owner=user)
         )
     )
+
+
+@pytest.fixture()
+def prevent_symlinking(monkeypatch: pytest.MonkeyPatch):
+    monkeypatch.setattr(
+        'cylc.flow.pathutil.make_symlink_dir',
+        lambda *_, **__: {}
+    )

--- a/tests/unit/test_install.py
+++ b/tests/unit/test_install.py
@@ -14,7 +14,6 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-import logging
 import os
 import shutil
 from pathlib import Path
@@ -30,7 +29,6 @@ from typing import (
 
 import pytest
 
-from cylc.flow import workflow_files
 from cylc.flow.exceptions import (
     InputError,
     WorkflowFilesError,
@@ -82,7 +80,8 @@ def test_install_workflow__max_depth(
     err_expected: bool,
     tmp_run_dir: Callable,
     tmp_src_dir: Callable,
-    glbl_cfg_max_scan_depth: NonCallableFixture
+    glbl_cfg_max_scan_depth: NonCallableFixture,
+    prevent_symlinking,
 ):
     """Test that trying to install beyond max depth fails."""
     tmp_run_dir()
@@ -109,7 +108,8 @@ def test_install_workflow__next_to_flow_file(
     flow_file: Optional[str],
     expected_exc: Optional[Type[Exception]],
     tmp_run_dir: Callable,
-    tmp_src_dir: Callable
+    tmp_src_dir: Callable,
+    prevent_symlinking,
 ):
     """Test that you can't install into a dir that contains a workflow file."""
     # Setup

